Did some changes in methods

AntonLubushko opened the pull request:
Added "syncronized" and try with resourses in some methods.

AntonLubushko committed:
Did some changes in methods

AntonLubushko commented:
Evgenyi Kondrashev gave me this issue, to look how I solve it.

AntonLubushko committed:
Added tests

AntonLubushko committed:
Realized Decorator pattern and refactor tests

AntonLubushko committed:
change in list of files

AntonLubushko committed:
refactor in names of classes and methods

ekondrashev commented on src/Parser.java:
```diff
@@ -0,0 +1,11 @@
+/**
+ * Created by Антон on 04.08.2016.
+ */
+public interface Parser {
```
Something is wrong with the name, it says `Parser` but does parse nothing. It just reads/saves..

ekondrashev commented on src/DefaultParser.java:
```diff
@@ -0,0 +1,41 @@
+        this.file = file;
+    }
+
+    public File getFile() {
```
Why we need this getter?

AntonLubushko committed:
removed getter, changed name of interface

AntonLubushko committed:
changed name of classes and interface

AntonLubushko committed:
changed name of interface

AntonLubushko committed:
another one change of classes names

AntonLubushko committed:
File -> Document

AntonLubushko committed:
Rename classes, involve "final", exclude setter

AntonLubushko committed:
shorted names of methods and classes

AntonLubushko committed:
String->CharSequence

AntonLubushko committed:
parse()->read()

AntonLubushko committed:
Parsable -> Readable

AntonLubushko committed:
method save() added

AntonLubushko committed:
Readable->Document
old Document->EntireDocument
Uses RandomAccessFile to lock a file while reading or writing

AntonLubushko committed:
EntireDocument->FileDocument
save()->write()
Uses Files.class to read/write
deleted StringBuilder instance

AntonLubushko committed:
Renamed test methods,
added some test methods,
 lock will be realised
 resources will be closed

AntonLubushko committed:
lock   in try-with-resources to autoclose
